src/envoy/utils/http_header_utils.cc

```diff
@@ -38,13 +38,17 @@ absl::string_view extractHeader(const Envoy::Http::HeaderMap& headers,
   return readHeaderEntry(entry);
 }
 
-absl::string_view getRequestHTTPMethodWithOverride(
-    absl::string_view originalMethod, const Envoy::Http::HeaderMap& headers) {
+bool handleHttpMethodOverride(Envoy::Http::HeaderMap& headers) {
   const auto* entry = headers.get(kHttpMethodOverrideHeader);
   if (entry) {
-    return entry->value().getStringView();
+    // Move the header.
+    headers.remove(Envoy::Http::Headers::get().Method);
```

Comment:
You can use build-in method:  headers.removeMethod()

Reply:
Ah thanks. The catch is you need to use `RequestHeaderMap` instead of the generic `HeaderMap` to make use of these `O(1)` operations. I can do that.